### PR TITLE
Add ConditionWithTimeMetNet3 for lead time conditioning (part of #55)

### PR DESCRIPTION
**Pull request by trudramukerji14**
## Description

Adds `ConditionWithTimeMetNet3`, the FiLM conditioning module for MetNet-3 lead time conditioning. This is the second step toward implementing MetNet-3 (see #[55](https://github.com/openclimatefix/metnet/issues/55)). This produces the scales and biases that will be consumed by the LeadTimeConditioner already [present](https://github.com/openclimatefix/metnet/blob/main/metnet/layers/LeadTimeConditioner.py) in the codebase when the full MetNet-3 model is wired together.

This paper cites the use of this as: "Lead Time Conditioning We use additive-multiplicative conditioning (FiLM, [13]) on lead time throughout the network. The conditioning is applied to the the network inputs and after each layer normalization.
All additive and multiplicative factors are outputted by a single MLP with one hidden layer of size 32 which
takes as input one-hot encoded lead time. The second layer of this MLP is initialized so that at initialization
the conditioning is an identity function" on in Appendix B. 

This implementation is similar to [ConditionWithTimeMetNet2](https://github.com/openclimatefix/metnet/blob/main/metnet/layers/ConditionWithTimeMetNet2.py) but some
key differences are:
- Hidden layer size of 32 (vs 2048 in MetNet-2), as specified in the MetNet-3 paper appendix
- Forecast steps of 722 (0 to 24 hours at 2 minute intervals, vs 720 in MetNet-2)
- Identity initialization on the second linear layer so conditioning starts as a no-op at initialization
- Intended to be used as a single shared instance across the whole model (vs one instance per layer in MetNet-2)

Fixes part of #61, part of #55

My future plan is continue working through the steps outlined in #55 

## How Has This Been Tested?

Added `test_condition_with_time_metnet3` to `tests/test_layers.py` which verifies:
- Output shapes are correct: `(batch, num_feature_maps)` for both scale and bias
- Identity initialization is correct: scale ≈ 1 and bias ≈ 0 at initialization

Run with:
```bash
uv run pytest tests/test_layers.py -x
```

image

- [x] Yes

## Checklist:
- [x] My code follows [OCF's coding style guidelines](https://github.com/openclimatefix/.github/blob/main/coding_style.md)
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

**Comment by trudramukerji14** [comment]
@jacobbieker This is a stab at next step of implementing MetNet3. I plan to work through the next steps of the outline. I think this will come together more cleanly when the model is actually wired together. But I wanted to but this out to get a feel for the repo's practice. Please feel free to provide any feedback.

File: tests/test_layers.py [hunk 1]
```diff
@@ -4,9 +4,27 @@
 from metnet.layers.MBConv import MBConv
 from metnet.layers.MultiheadSelfAttention2D import MultiheadSelfAttention2D
 from metnet.layers.PartitionAttention import BlockAttention, GridAttention
+from metnet.layers.ConditionWithTimeMetNet3 import ConditionWithTimeMetNet3
+
 import torch
 
 
+def test_condition_with_time_metnet3():
```
Comment (jacobbieker) [inline tests/test_layers.py:13]: Can you add a test with a backward pass, to ensure gradients are flowing?

**Review by jacobbieker — CHANGES_REQUESTED** [review]
Looks good! Thanks for this, only thing I think needs to be added is a test to ensure gradients work.

File: tests/test_layers.py (hunk 1 above)
Comment (trudramukerji14) [reply tests/test_layers.py:13]: Sure will do. Thank you for the feedback!

Comment (trudramukerji14) [reply tests/test_layers.py:13]: I added a test for the backward pass with `test_backward_pass_gradients_flow`

The test instead verifies that all parameters have defined gradients but please do let me know if I missed something!

**Review by jacobbieker — APPROVED** [review]
LGTM! Thanks!

File: tests/test_layers.py (hunk 1 above)
Comment (trudramukerji14) [reply tests/test_layers.py:13]: image
